drivers/vulkan/vulkan_context.cpp

```diff
@@ -2765,7 +2767,13 @@ void VulkanContext::set_vsync_mode(DisplayServer::WindowID p_window, DisplayServ
 	_update_swap_chain(&windows[p_window]);
 }
 
-VulkanContext::VulkanContext() {
+VulkanContext::VulkanContext(const uint8_t p_frame_count, const uint8_t p_swapchain_count) {
```

Comment:
Is there a reason to pass these values through the Context creation? You should be able to read the project settings here using ``GLOBAL_GET`` (or alternatively, you could fill these values in initialize as well)

If I understand correctly, the settings need to be created and validated in main.cpp. But I think we could rely on the project setting system to read them here instead of adding plumping throughout the displayserver

Reply:
```suggestion
VulkanContext::VulkanContext(uint8_t p_frame_count, uint8_t p_swapchain_count) {
```
Match the declaration

Reply:
> Is there a reason to pass these values through the Context creation? You should be able to read the project settings here using GLOBAL_GET (or alternatively, you could fill these values in initialize as well)

No particular reason, I was unsure if I could get the GLOBAL_GET correctly when called afterwards, so I just played safe.

If you tell me it's fine to call `GLOBAL_GET` from the VulkanContext then great, I didn't like passing that plumbing either.

> Since these can only be changed at startup, they should use GLOBAL_DEF_RST_BASIC which will prompt the user to restart if they adjust the setting.
>
> Personally, I think these should be GLOBAL_DEF_RST as they are settings for advanced users only. Removing BASIC means that they setting will only appear if the "advanced" toggle is toggled on.

Ah! Awesome.